tests/e2e-openshift/monitoring/chainsaw-test.yaml

```diff
@@ -0,0 +1,44 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/kyverno/chainsaw/main/.schemas/json/test-chainsaw-v1alpha1.json
+apiVersion: chainsaw.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  creationTimestamp: null
+  name: monitoring
+spec:
+  steps:
+  - name: step-00
+    try:
+    - apply:
+        file: 00-install-storage.yaml
+    - assert:
+        file: 00-assert.yaml
+  - catch:
+    - script:
+        content: ./tests/e2e-openshift/monitoring/check_user_workload_monitoring.sh
+    name: step-01
+    try:
+    - apply:
+        file: 01-workload-monitoring.yaml
+  - name: step-02
+    try:
+    - apply:
+        file: 02-install-tempostack.yaml
+    - assert:
+        file: 02-assert.yaml
+  - name: step-03
+    try:
+    - apply:
+        file: 03-generate-traces.yaml
+    - assert:
+        file: 03-assert.yaml
+  - name: step-04
+    try:
+    - apply:
+        file: 04-verify-traces.yaml
+    - assert:
+        file: 04-assert.yaml
+  - catch:
+    - script:
+        content: ./tests/e2e-openshift/monitoring/check_metrics.sh
```

Comment:
This doesn't work. I was testing the opentelemetry-operator tests and the tests referencing the scripts are failing. We need to specify the script content instead of the path. https://kyverno.github.io/chainsaw/latest/operations/script/ Also the scripts that are used as assert in KUTTL needs to be modified such that it keeps running until the script is successful or timeout is reached.

Reply:
I'm working on a PR to fix the OpenTelemetry Operator tests, also to get them working with OpenShift CI.

Reply:
@IshwarKanse did it work before ? To me scripts have always been relative to the test folder 🤔

Reply:
> the scripts that are used as assert in KUTTL needs to be modified such that it keeps running until the script is successful or timeout is reached

was KUTTL doing this ? i don't remember KUTTL retrying scripts in a loop.

Reply:
It did via the TestAssert https://github.com/grafana/tempo-operator/blob/main/tests/e2e-openshift/monitoring/05-assert.yaml https://kuttl.dev/docs/testing/reference.html#testassert

Reply:
Sure but `./tests/e2e-openshift/monitoring/check_metrics.sh` is not a valid relative path from the test folder

Reply:
In chainsaw (and KUTTL i believe) all paths are relative to the test folder, not the working directory from which the tool is invoked.

Reply:
```yaml
    - script:
        timeout: 1m
        content: ./tests/e2e-openshift/monitoring/check_metrics.sh
```

Reply:
There is inconsistency in KUTTL on the path reference. With TestStep, it uses relative path. With TestAssert it uses a path from the directory where kuttl is run from like the root of the operator repo. It is a mess that we had to live with. haha

Reply:
Doh, i will fix the chainsaw command that converts kuttl tests to chainsaw to account for that 🙈 
I learn KUTTL gotchas every day... chainsaw is definitely more simple to reason about, everything is relative to the test forlder.

Sorry for not catching it (i confess that i didn't run the openshift tests, not comfortable enough with openshift).

Reply:
Ok so relative path for scripts are working in Chainsaw. I was testing with this. 

        === COMMAND
        /bin/sh -c ./check_traces.sh
    | 16:39:51 | kafka | step-05  | SCRIPT    | LOG   |
        === STDOUT
        "-> service.name: Str("kafka")" found in kafka-receiver-collector-6f84bfdcf-4k9h4
        "-> test: Str(kuttl-kafka)" found in kafka-receiver-collector-6f84bfdcf-4k9h4
        Traces with service name Kafka and attribute test=kuttl-kafka found.
    | 16:39:51 | kafka | step-05  | SCRIPT    | DONE  |
    | 16:39:51 | kafka | step-05  | TRY       | DONE  |
--- PASS: chainsaw (0.00s)
    --- PASS: chainsaw/kafka (47.44s)
PASS
Tests Summary...
- Passed  tests 1
- Failed  tests 0
- Skipped tests 0
Done.

  - name: step-05
    try:
    - script:
        content: ./check_traces.sh
        
It would be cool if we do not have to specify ./ if the file is in the same directory. Also we have to add the executable permission to the script with chmod +x so that it can run.